fw/sock.c

```diff
@@ -904,6 +905,22 @@ do {									\
 			continue;
 		}
 
+		if (unlikely(skb_headroom(skb) < MAX_TCP_HEADER)) {
+			struct sk_buff *twin_skb =
+				__pskb_copy_fclone(skb, MAX_TCP_HEADER,
+						   GFP_ATOMIC, true);
+			if (!twin_skb) {
+				T_WARN("Unable to copy SKB with small "
+				       "headroom.\n");
+				__kfree_skb(skb);
+				r = SS_BAD;
+				goto out;
+			}
```

Comment:
Why don't we fix it in `MAX_HEADER`, like we do this for TLS headers, i.e. just increase the allocation overhead for all allocated `skb`s?

This is `unlikely` specifier, but when this actually happen? `ss_tcp_process_skb()` processes all the ingress packets, so it is super hot and we should no introduce skb reallocations here. Moreover, in most of the cases we do no retransmit skbs and are just going to introduce the overhead.